apps/kyb-app/src/components/organisms/DynamicUI/TransitionListener/TransitionListener.tsx

```
@@ -11,37 +12,49 @@
-    prevStateRef.current = state;
+    if (currentPageIndex !== -1) {
+      prevStateRef.current = state;
+    }
   }, [prevStateRef, helpersRef, state, pages]);
 
-  return <>{children}</>;
+  useEffect(() => {
+    const currentPageIndex = pages.findIndex(page => page.stateName === state);
+
+    if (currentPageIndex === -1 && state === 'finish') {
+      onFinish && onFinish({ setElementCompleted: helpersRef.current }, prevStateRef.current);
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)